### PR TITLE
[pickers] Rename `ClockPicker` into `TimeClock`

### LINKED ISSUE #6709: [pickers] Rename `ClockPicker` into `TimeClock`
## Why ?

The word *picker* now only describes a component with both a field for keyboard editing and a set of views for UI editing.

Side note: the static pickers will probably evolve in the future to fit this nomenclature

### PR DESCRIPTION
**Pull request by flaviendelangle**
Closes #6709

#### Breaking changes

The `ClockPicker` view component has been renamed  `TimeClock` to better fit its usage:

```diff
-<ClockPicker {...props} />
+<TimeClock {...props} />
```

Component name in the theme have changed as well:

```diff
-MuiClockPicker: {
+MuiTimeClock: {
```

### CODE REVIEW

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 1]
```diff
@@ -192,31 +192,26 @@ The `locale` prop of the `LocalizationProvider` component have been renamed `ada
 
 ### Rename the view components
 
-The view components allowing to pick a date or parts of a date without an input have been renamed to better fit their usage:
+The view components allowing to pick a time, a date or parts of a date without an input have been renamed to better fit their usage:
 
 ```diff
```
Comment (flaviendelangle) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:197]: Slightly out of topic, this section was the only one with splitted diffs

File: docs/data/migration/migration-pickers-v5/migration-pickers-v5.md [hunk 2]
```diff
@@ -236,24 +231,27 @@ Component names in the theme have changed as well:
...
 +MuiYearCalendar: {
+
+-MuiClockPicker: {
++MuiTimeClock: {
 ```
 
 ### Rename `date` prop to `value` on view components
 
-The `date` prop has been renamed `value` on `MonthPicker`, `YearPicker`, `ClockPicker` and `CalendarPicker`:
+The `date` prop has been renamed `value` on `MonthCalendar`, `YearCalendar`, `TimeClock` and `DateCalendar`:
```
Comment (flaviendelangle) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:241]: I'm using the new component names here, I take for granted that people read the section above
What do you think ?

Comment (flaviendelangle) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:241]: I'm using the new component names here, I take for granted that people read the section above
What do you think ?

Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v5/migration-pickers-v5.md:241]: You can insist a bit if people are reading it without doing the modification step by step
```suggestion
The `date` prop has been renamed `value` on `MonthCalendar`, `YearCalendar`, `TimeClock` and `DateCalendar` (components renamed in previous section):
```

**Review by alexfauquette — APPROVED** [review]

**Pull request merged**